### PR TITLE
Minor changes to support e2e testing

### PR DESCRIPTION
**Pull request by sethvargo**

### CODE REVIEW

File: internal/authorizedapp/memory_provider.go [hunk 1]
```diff
@@ -27,26 +29,44 @@ var _ Provider = (*MemoryProvider)(nil)
 // MemoryProvider is an Provider that stores values in-memory. It is primarily
 // used for testing.
 type MemoryProvider struct {
-	Data map[string]*model.AuthorizedApp
+	lock sync.RWMutex
+	data map[string]*model.AuthorizedApp
```
Comment (mikehelmick) [inline internal/authorizedapp/memory_provider.go:33]: this is just for tests right, why not have tests use the test DB infrastructure?

File: internal/authorizedapp/memory_provider.go [hunk 2]
```diff
@@ -27,26 +29,44 @@ var _ Provider = (*MemoryProvider)(nil)
...
 }
 
 // NewMemoryProvider creates a new Provider that reads from a database.
```
Comment (mikehelmick) [inline internal/authorizedapp/memory_provider.go:36]: comment out of date

File: internal/authorizedapp/memory_provider.go (hunk 1 above)
Comment (sethvargo) [reply internal/authorizedapp/memory_provider.go:33]: This is faster and cheaper.

**Pull request merged**